### DISCUSSION TITLE
SEPA reason for payment

### DISCUSSION THREAD

**Discussion by anongit111** [bisq-network/bisq#6764] [Ideas]
Is it possible to add a checkbox which tells the taker to use the make/taker account name as reason for payment?

I ask because it seems that many banks auto create a reason for payment if left blank. I as maker don't want strange reason for payment.

**Comment by pazza83** [comment]
Most trades seem to be completed with references being left blank. When a reference is required the wiki recommends to use buyers account name.

> If you are required to fill in a reason, use your account name eg "Joe Bloggs" alternatively enter something non-descriptive like a dash "-". If you would prefer to use other text please agree this with your peer in trader chat. Please remember your peer is not obligated to agree anything and may not want to agree to anything other than leaving the description blank, your name or a dash.

https://bisq.wiki/Trading_rules#Leave_the_.22reason_for_payment.22_field_empty

**Reply by anongit111** [reply]
I understood that and i know . But as a maker sometimes the payment has started before i can chat them to use the account name . I'd be even happy with a script that chats when a trade is accepted . Would be open to offer 1 Mil. sats for a solution .

**Comment by anongit111** [comment]
I'm hereby increasing my offer to two million sats for a solution . Escrow by a bisq maintainer preferred .

**Comment by ghost** [comment]
>  as a maker sometimes the payment has started before i can chat them

This looks like a useful feature: the ability for maker to define a greeting chat message sent to the taker immediately as the trade starts.  It is certainly doable, I have already have an experimental proof-of-concept.  Greeting messages could be customized as either one fits all or specific per-currency / per payment method.

Would like to get other team reactions before committing further.

**Reply by suddenwhipvapor** [reply]
adding a human touch to a trading system that is not the most welcoming to newbie users will be an appreciated feature, IMO

**Comment by pazza83** [comment]
I have some concerns about adding a defined opening chat message for traders.

The reason is I think the payment account set up should give the buyer's everything they need to make the payment.

There are cases where traders can 'agree' to use a specific reference or for the payment to be made a certain way. But having a message sent automatically by a trader might give a request the appearance of authority to a new buyer.

For example messages like:

"Hi thanks for taking this trade please make payment with a reference related to the purchase of an electronic good."
"Hi thanks for taking this trade. When you pay please add your name as the payment reference"
 
Whilst it is fine to request these things they should be in agreement with the buyer and they should be able to refuse and send the payment with no reference if they prefer.

The buyer could always message manually but the nature of Bisq at least means that the makers of the offer are not always present when the trade is taken, meaning they should expect to receive payment as per the usual trading protocols.

> I ask because it seems that many banks auto create a reason for payment if left blank. I as maker don't want strange reason for payment.

If someone was to send a payment with a strange reference this is already against trade rules and could be punished.

**Comment by anongit111** [comment]
And if the taker leaves reason for payment blank and the bank adds a strange reference ? Plausible deniability right ?
> If someone was to send a payment with a strange reference this is already against trade rules and could be punished.

As a maker i don't wait for takers 'til they take an offer . How am i supposed to tell them to use the name as a reference when the payment has already started ?

**Reply by pazza83** [reply]
I think my point would be that you do not. Buyers will send payment to you with no reference as in the trade guidelines.

- Say every buyer ended up using their name as the reference when buying from you, I fear that could make it easier for Bisq traders to identified as they all share a similar pattern when it comes to a given reference.

**Reply by anongit111** [reply]
Well that works both ways . And reason left blank does not reference a similar pattern ?
And i like @jmacxx idea of a greeting message . I'm still offering a 2 mil sat reward . Doesn't have to be a official Bisq solution .

**Reply by ghost** [reply]
@pazza83 I think sending a chat message is ok.  Rules are displayed when opening trader chat.  I don't think sending a chat message automatically conveys any authority.

I would like to implement this for @anongit111's use and claim the bounty.
It can be provided as a patch which they would need to [build](https://github.com/bisq-network/bisq/blob/master/docs/build.md).

Trying to be polite and upfront about this.  Really don't see any problem with the request.

**Reply by ghost** [reply]
@anongit111 could you contact me through email (on my github profile).

**Reply by anongit111** [reply]
Mail sent . I use a trashmailer . Maybe check spam . Do you nostr ?

**Reply by ghost** [reply]
@anongit111 Did not receive email.  Yes, **npub** in my profile.  Could you do the same?  Thanks.

---

![image](https://github.com/bisq-network/bisq/assets/47253594/aa32483d-1692-45af-80c2-9bc51eab1116)


![image](https://github.com/bisq-network/bisq/assets/47253594/c73b8339-1158-4842-9cbc-0995eef25e9d)

**Comment by ghost** [comment]
@anongit111 was happy with the feature and has asked if it can be added to the official Bisq build : [PR opened](https://github.com/bisq-network/bisq/pull/6788).

**Comment by HenrikJannsen** [comment]
I do not think it is a good idea to add such a popup. With anything additional we make UX more heavy and ppl are complaining already that Bisq is too complicate and too much to read/learn.
I do not see the need for that feature and that the added value justify the added mental costs for  the users.
Beside that I consider the design not adequate.
